### PR TITLE
Enable hackage-tests on downstream travis

### PR DESCRIPTION
**Pull request by phadej**
Please include the following checklist in your PR:

* [ ] Patches conform to the [coding conventions](https://github.com/haskell/cabal/#conventions).
* [ ] Any changes that could be relevant to users have been recorded in the changelog.
* [ ] The documentation has been updated, if necessary.
* [ ] If the change is docs-only, `[ci skip]` is used to avoid triggering the build bots.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

File: travis/binaries/travis-test.sh [hunk 1]
```diff
@@ -21,9 +21,8 @@ ln -s $TRAVIS_BUILD_DIR $UPSTREAM_BUILD_DIR
 
 # Test we can parse Hackage
 # Note: no $TEST_OPTIONS as this isn't tasty suite
-if [ -f ~/.cabal/config ]; then
+cabal update # fetch 01-index.tar
```
Comment (23Skidoo) [inline travis/binaries/travis-test.sh]: I see, that was a temporary band-aid. Can we do `user-config init` here instead? That should be faster.

Comment (phadej) [reply travis/binaries/travis-test.sh]: We need `update`, `hackage-tests` need index file.

Comment (phadej) [reply travis/binaries/travis-test.sh]: ... should I add that as a comment?

Comment (23Skidoo) [reply travis/binaries/travis-test.sh]: No harm in that.

**Pull request merged**